crates/hotshot/src/demo.rs

```diff
@@ -66,11 +62,11 @@ impl Default for SDemoState {
 impl State for SDemoState {
     type Error = BlockPayloadError;
 
-    type BlockType = VIDBlockPayload;
+    type BlockPayload = VIDBlockPayload;
 
     type Time = ViewNumber;
 
-    fn validate_block(&self, _block: &Self::BlockType, view_number: &Self::Time) -> bool {
+    fn validate_block(&self, _block: &Self::BlockPayload, view_number: &Self::Time) -> bool {
```

Comment:
I think this should also take the header, which could have metadata important for interpreting the payload (e.g. namespace table). Also, the payload should eventually be replaced by the "validated state", but...we'll get to that later